### ISSUE TITLE
[java] UnnecessaryBoxing false positive in PMD 7.0.0 in lambda

### ISSUE THREAD

**Issue by julius-d** [pmd/pmd#4924]
Hi!
I am currently trying to update from 6.55.0 to PMD 7.0.0.

**Affects PMD Version:** 7.0.0

**Rule:** [UnnecessaryBoxing](https://pmd.github.io/pmd/pmd_rules_java_codestyle.html#unnecessaryboxing)

**Code Sample demonstrating the issue:**

```java
import java.util.Optional;

public class Example {

  record Item(int cents) {}

  Long map(Item item) {
    return Optional.ofNullable(item).map(it -> Long.valueOf(it.cents())).orElse(null);
  }
}
```

**Outcome:**
```
PMD Failure: Example:10 Rule:UnnecessaryBoxing Priority:3 
Unnecessary explicit conversion from int to Function<? super Item, ? extends Long>
```

**Running PMD through:** *maven-pmd-plugin 3.21.2*
